.github/workflows/docker_build_push.yml

```
@@ -182,18 +194,32 @@ jobs:
           context: ${{ inputs.docker_build_context }}
           push: ${{ !inputs.skip_image_push }}
           ssh: ${{ inputs.docker_buildx_driver == 'docker-container' && env.SSH_DEFAULT || env.SSH_AUTH_SOCK }}
-          build-args: "GIT_COMMIT=${{ github.sha }}"
+          build-args: "GIT_COMMIT=${{ inputs.github_ref || github.sha }}"
           secret-files: BUILD_KEY=/tmp/build_key.json
           tags: |
-            "${{ env.IMAGE_NAME }}:${{ github.sha }}"
-            ${{ github.event_name == 'pull_request' && format('{0}:pr-{1}', env.IMAGE_NAME, github.event.number) || '' }}
-            ${{ github.ref == format('refs/heads/{0}', inputs.branch) && format('{0}:latest', env.IMAGE_NAME) || '' }}
-            ${{ github.ref == format('refs/heads/{0}', inputs.branch) && format('{0}:stage', env.IMAGE_NAME) || '' }}
+            "${{ env.IMAGE_NAME }}:${{ inputs.github_ref || github.sha }}"
           provenance: false
 
-      # if either a tag is provided, run re-tag the image with prod and the commit hash
-      - if: startsWith(github.ref, 'refs/tags/')
-        name: Set release tags
+      - if: ${{ github.ref == 'refs/heads/main' && (!inputs.github_ref || inputs.github_ref == github.sha) }}
```

`inputs.branch` is needed since trunk could be `main`, `master` or whatever branch a user wants to tag stage with
```suggestion
      - if: ${{ github.ref == format('refs/heads/{0}', inputs.branch) && !inputs.latest_production_tags }}
```